### DISCUSSION TITLE
Content negotiation other than accept-encoding in the wild

### DISCUSSION THREAD

**Discussion by ibnesayeed** [cs531-f22/cs531-f22.github.io#19] [Assignment]
Find up to three different "in the wild" examples of content negotiation in dimensions other than `Accept-Encoding`.

**Comment by rohitrahiya** [comment]
Hi Professor, here I have mentioned examples of content negotiation other than  Accept-Encoding.

$ curl -IL -A "Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" https://www.pogo.com | grep -iE "^(http|location|Vary)"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0 57156    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/1.1 200 OK
Vary: Origin
Vary: Access-Control-Request-Method
Vary: Access-Control-Request-Headers

 
 

$ curl -IL -A "Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" https://www.hulu.com | grep -iE "^(http|location|Vary)"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   142    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
HTTP/2 302
location: https://www.hulu.com/welcome
vary: Origin
HTTP/2 200
vary: Origin

 
 
$ curl -IL -A "Mozilla/5.0 (iPhone; CPU iPhone OS 12_2 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148" https://www.swiggy.com | grep -iE "^(http|location|Vary)"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/2 200
vary: Origin

**Comment by Badjedi04** [comment]
**Hello Professor, Below are the examples of content negotiation other than Accept-Encoding**

' ' ' '
$ curl -IL https://www.skywest.com
HTTP/2 200 
date: Thu, 20 Oct 2022 03:05:08 GMT
content-type: text/html; charset=utf-8
set-cookie: AWSALB=SD3X6GZKpe2RAMK6KX5iEA26qZPg2rmfpXQLNOOKnO2oZt9loIwSHT668cv+uWn27dgW6q4BZ5qUlzO7wss7F2oNnglrgO+/qhwlnps31ifRsU5YVus3kU8g4RVE; Expires=Thu, 27 Oct 2022 03:05:07 GMT; Path=/
set-cookie: AWSALBCORS=SD3X6GZKpe2RAMK6KX5iEA26qZPg2rmfpXQLNOOKnO2oZt9loIwSHT668cv+uWn27dgW6q4BZ5qUlzO7wss7F2oNnglrgO+/qhwlnps31ifRsU5YVus3kU8g4RVE; Expires=Thu, 27 Oct 2022 03:05:07 GMT; Path=/; SameSite=None; Secure
server: Apache/2.4.54 () OpenSSL/1.0.2k-fips
content-security-policy: default-src 'self'; script-src 'self' 'unsafe-inline' 'unsafe-eval' https://www.googleadservices.com https://chimpstatic.com https://www.google-analytics.com https://siteimproveanalytics.com https://www.googletagmanager.com https://www.google.com https://snap.licdn.com https://connect.facebook.net https://googleads.g.doubleclick.net https://www.gstatic.com https://public.tableau.com; connect-src 'self' https://www.google-analytics.com https://stats.g.doubleclick.net https://www.googletagmanager.com; img-src 'self' https://i.ytimg.com https://www.linkedin.com https://files-skywest-com.s3.us-west-2.amazonaws.com https://*.siteimproveanalytics.io https://googleads.g.doubleclick.net https://www.facebook.com https://www.google.com https://px.ads.linkedin.com https://www.google-analytics.com https://www.skywest.com https://placehold.it https://www.placeholder.com https://placeholder.com https://public.tableau.com data:; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com https://cdnjs.cloudflare.com; base-uri 'self'; form-action 'self' https://www.facebook.com; frame-src 'self' https://www.google.com https://bid.g.doubleclick.net https://www.youtube.com https://youtube.com https://public.tableau.com https://www.facebook.com; font-src 'self' https://fonts.gstatic.com https://cdnjs.cloudflare.com
**vary: X-Forwarded-Protocol**
cache-control: private, must-revalidate
set-cookie: PHPSESSID=10d0qan1vvb0lgtqg2j09o9n2t; path=/; HttpOnly
etag: "a86e7e65e6d75a8fc640b340168784ef"
strict-transport-security: max-age=10886400;
access-control-allow-credentials: true
access-control-allow-methods: GET, PUT, POST, DELETE, OPTIONS
access-control-max-age: 1000
access-control-allow-headers: Origin, Content-Type, X-Auth-Token , Authorization
set-cookie: visid_incap_2753890=ogW1Dv1ISpmRhQHxJap5meO6UGMAAAAAQUIPAAAAAAC67sMtfnEamzMpLlP+SLCR; expires=Thu, 19 Oct 2023 07:23:23 GMT; HttpOnly; path=/; Domain=.skywest.com
set-cookie: nlbi_2753890=hIKbatbO4l8DsRdF7fV2QwAAAAAxKmRm4SpaOxXALu/Z7o2o; path=/; Domain=.skywest.com
set-cookie: incap_ses_1349_2753890=0ID8AA/MmXzWiGdS6Jy4EuO6UGMAAAAAUyzQFDapdaeFi9lfAmg5Jw==; path=/; Domain=.skywest.com
x-cdn: Imperva
x-iinfo: 10-13284680-13284697 NNNN CT(61 124 0) RT(1666235106580 148) q(0 0 2 -1) r(9 9) U6

' ' ' '

' ' ' '
$ curl -IL https://www.europa.eu
HTTP/1.1 302 Found
Location: https://europa.eu/
Cache-Control: no-cache
Pragma: no-cache
X-XSS-Protection: 1
Content-Type: text/html; charset=utf-8
Proxy-Connection: close
Connection: close
Content-Length: 683

HTTP/1.1 301 Moved Permanently
Date: Thu, 20 Oct 2022 03:15:45 GMT
Content-Type: text/html; charset=iso-8859-1
Location: https://european-union.europa.eu/
Age: 0
X-FPFIS: 761731453
Server: Europa
Proxy-Connection: Keep-Alive
Connection: Keep-Alive

HTTP/2 302 
content-type: text/html; charset=UTF-8
content-length: 0
date: Thu, 20 Oct 2022 03:15:45 GMT
cache-control: no-cache, private
x-content-type-options: nosniff
location: /select-language?destination=/node/1
x-fpfis: 1049549 327683
strict-transport-security: max-age=31536000; includeSubDomains; preload;
x-cache: Miss from cloudfront
via: 1.1 01b8e3a9ba527e97cf26e596dffabb8e.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD55-P5
x-amz-cf-id: 8nO6pukhIIqHriSfOW9-WbCHF_JXM9Xww-mVJTdtNsvwA409gpC80Q==
age: 1539

HTTP/2 200 
content-type: text/html; charset=UTF-8
content-length: 0
date: Thu, 20 Oct 2022 03:15:45 GMT
cache-control: max-age=21600, public
x-ua-compatible: IE=edge
content-language: en
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
permissions-policy: interest-cohort=()
expires: Sun, 19 Nov 1978 05:00:00 GMT
last-modified: Wed, 19 Oct 2022 20:58:23 GMT
etag: "1666213103"
**vary: Cookie**
accept-ranges: bytes
x-fpfis: 756811 131075
strict-transport-security: max-age=31536000; includeSubDomains; preload;
x-cache: Miss from cloudfront
via: 1.1 01b8e3a9ba527e97cf26e596dffabb8e.cloudfront.net (CloudFront)
x-amz-cf-pop: IAD55-P5
x-amz-cf-id: NBiVHl-GvczT7P_HI0uxW9ZrlB9rvfhL3ZNDv--pU6ZFOF5CGu_blA==
age: 1539

' ' ' '

' ' ' '
$ curl -IL https://www.github.com
HTTP/2 301 
content-length: 0
location: https://github.com/

HTTP/2 200 
server: GitHub.com
date: Thu, 20 Oct 2022 03:18:06 GMT
content-type: text/html; charset=utf-8
**vary: X-PJAX, X-PJAX-Container, Turbo-Visit, Turbo-Frame, Accept-Language, Accept-Encoding, Accept, X-Requested-With**
content-language: en-US
etag: W/"17bd21b776c0e8d30a7664514970b1de"
cache-control: max-age=0, private, must-revalidate
strict-transport-security: max-age=31536000; includeSubdomains; preload
x-frame-options: deny
x-content-type-options: nosniff
x-xss-protection: 0
referrer-policy: origin-when-cross-origin, strict-origin-when-cross-origin
expect-ct: max-age=2592000, report-uri="https://api.github.com/_private/browser/errors"
content-security-policy: default-src 'none'; base-uri 'self'; block-all-mixed-content; child-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/; connect-src 'self' uploads.github.com objects-origin.githubusercontent.com www.githubstatus.com collector.github.com raw.githubusercontent.com api.github.com github-cloud.s3.amazonaws.com github-production-repository-file-5c1aeb.s3.amazonaws.com github-production-upload-manifest-file-7fdce7.s3.amazonaws.com github-production-user-asset-6210df.s3.amazonaws.com cdn.optimizely.com logx.optimizely.com/v1/events *.actions.githubusercontent.com wss://*.actions.githubusercontent.com online.visualstudio.com/api/v1/locations github-production-repository-image-32fea6.s3.amazonaws.com github-production-release-asset-2e65be.s3.amazonaws.com insights.github.com wss://alive.github.com github.githubassets.com; font-src github.githubassets.com; form-action 'self' github.com gist.github.com objects-origin.githubusercontent.com; frame-ancestors 'none'; frame-src viewscreen.githubusercontent.com notebooks.githubusercontent.com; img-src 'self' data: github.githubassets.com identicons.github.com github-cloud.s3.amazonaws.com secured-user-images.githubusercontent.com/ github-production-user-asset-6210df.s3.amazonaws.com customer-stories-feed.github.com spotlights-feed.github.com *.githubusercontent.com; manifest-src 'self'; media-src github.com user-images.githubusercontent.com/ secured-user-images.githubusercontent.com/ github.githubassets.com; script-src github.githubassets.com; style-src 'unsafe-inline' github.githubassets.com; worker-src github.com/assets-cdn/worker/ gist.github.com/assets-cdn/worker/
set-cookie: _gh_sess=473R7E4SURofSjIYFVj5sep1dXYOmbhafMpjAX4NBCr%2FqXv9gDPOtgBuTPrkGYnvuku0dH%2F8vXtBcnr8ZJ7ME2vNQjUvO63GHOksd8XiEGOOqZ7Rmrwk5F%2F4rgUoYOJn7ILbtrXRQgSbTMZ5pcoGoHaU79qeARw3nhCZmTMfGA1f70hmEvPR%2FtjO%2Fj2%2BS%2B%2Bf4X3Wwz6Cw3OgTp9p6FAERaVhwZYMaIiNnfewvF%2B0iynhHWTZMRga3pQg4hNd3Jslhvw7BKRsCqIkslmZtJYTmw%3D%3D--jNLyRs00SDYjVbSY--e1u06v%2Bex7sp7nvTLND66A%3D%3D; Path=/; HttpOnly; Secure; SameSite=Lax
set-cookie: _octo=GH1.1.683451992.1666235887; Path=/; Domain=github.com; Expires=Fri, 20 Oct 2023 03:18:07 GMT; Secure; SameSite=Lax
set-cookie: logged_in=no; Path=/; Domain=github.com; Expires=Fri, 20 Oct 2023 03:18:07 GMT; HttpOnly; Secure; SameSite=Lax
accept-ranges: bytes
x-github-request-id: F5B2:63CD:A31FB4:EADA99:6350BDEF

' ' ' '

**Comment by calebkbrad** [comment]
Here are my examples of content negotiation other than Accept-Encoding.

```
$ curl -IL https://instagram.com | grep -iE "^(http|location|Vary)"
HTTP/2 405
vary: Cookie, Accept-Language
```

```
$ curl -IL https://nbc.com | grep -iE "^(http|location|Vary)"
HTTP/1.1 301 Moved Permanently
Location: https://www.nbc.com/
HTTP/2 200
vary: User-Agent
```

```
$ curl -IL https://blackboard.com | grep -iE "^(http|location|Vary)"
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0   150    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
HTTP/2 301
location: https://www.blackboard.com/
HTTP/2 200
vary: Cookie,Accept-Encoding,Host,X-Geo-Country
```
